internal/xcuitest/config.go

```diff
@@ -226,3 +230,58 @@ func SortByHistory(suites []Suite, history insights.JobHistory) []Suite {
 	}
 	return res
 }
+
+// ShardSuites applies sharding by provided shard config.
+func ShardSuites(p *Project) error {
+	var suites []Suite
+	for _, s := range p.Suites {
+		if s.Shard != "concurrency" {
+			suites = append(suites, s)
+			continue
+		}
+		shardedSuites, err := parseTestClassesFile(s, p.Sauce.Concurrency)
+		if err != nil {
+			log.Warn().Err(err).Msgf("Failed to get tests from %q. Using default settings.", s.TestClassesFile)
+			suites = append(suites, s)
+			continue
+		}
```

Comment:
As per my [earlier comment](https://github.com/saucelabs/saucectl/pull/772#discussion_r1192636140), I really think this needs to fail. Otherwise the feedback loop is too slow for the user to address this issue (or is ignored by accident in a CI environment).